packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,129 @@ class _Location {
   }
 }
 
+/// Transformer to parse and gather information about [offendingContext].
+///
+/// This transformer only responds to [DiagnosticsProperty] with the name
+/// 'offendingContext'.
+///
+/// This function will be registered to [FlutterErrorDetails.informationTransformers]
+/// in [WidgetsBinding.initInstances].
+DiagnosticPropertiesBuilder transformOffendingContext(DiagnosticPropertiesBuilder properties) {
+  final DiagnosticPropertiesBuilder transformed = DiagnosticPropertiesBuilder();
+  bool foundStackTrace = false;
+  final List<DiagnosticsNode> afterStackTrace = <DiagnosticsNode>[];
+  // The below for loop try to achieve the following:
+  // 1. Any offendingContext found before the stacktrace stay where it is.
+  // 2. Any offendingContext found after the stacktrace move to the front of stacktrace.
+  for (DiagnosticsNode node in properties.properties) {
+    if (!foundStackTrace && node.runtimeType == DiagnosticsStackTrace)
+      foundStackTrace = true;
+    final Iterable<DiagnosticsNode> parsed = _parseDiagnosticsNode(node);
+    if (parsed != null) {
+      parsed.forEach(transformed.add);
+    }
+    else {
+      if (foundStackTrace) {
+        afterStackTrace.add(node);
+      } else {
+        transformed.add(node);
+      }
+    }
+  }
+  afterStackTrace.forEach(transformed.add);
+  return transformed;
+}
+
+/// Transform the input [DiagnosticsNode].
+///
+/// Return null if input [DiagnosticsNode] is not applicable.
+Iterable<DiagnosticsNode> _parseDiagnosticsNode(DiagnosticsNode node) {
+  if (node.name != 'offendingContext' || node is! DiagnosticsProperty<Object>)
```

Comment:
If we want to be deriving this from the name, then we should use a constant everywhere. Otherwise, we'll have (and miss) typos.

But I think we're probably better off just making a subclass of DiagnosticsProperty that you can recognize by type, or some such. For example, DiagnosticsOffendingContext or ErrorContext or something.